// nodejsforDB/dbConnect.js
var mysql = require('mysql');
function connectDBMysql(){
  var conn = mysql.createConnection({
      host: 'localhost',
      user: 'nodejs',
      password: 'nodejs',
      database: 'nodejs',
      port: 3306
  });
  return conn;

}
/***********添加数据表***************/
function creatdataTable(conn,tablename,callback){
  conn.query('CREATE TABLE '+tablename+' (MT varchar(255),MW varchar(255),OT varchar(255),OW varchar(255),LT varchar(255),SF varchar(255),O2 varchar(255))',function(err,result){
    if(err){
      console.log('[INSERT ERROR] - ',err.message);
      return err;
    }
    callback(err,result);
    console.log('-------CREATTABLE----------');
    console.log('CREATTABLE ID:',result);
  });

}
/***********插入数据***************/
function insertdataFun(conn,tablename,MT,MW,OT,OW,LT,SF,O2,callback){
  conn.query('INSERT INTO '+tablename+'(MT,MW,OT,OW,LT,SF,O2)'+' VALUES('+MT+','+MW+','+OT+','+OW+','+LT+','+SF+','+O2+')',function(err,result){
    if(err){
     console.log('[INSERT ERROR] - ',err.message);

     return err;
    }
    callback(err,result);
   console.log('-------INSERT----------');
  //  console.log('INSERT ID:',result.insertId);
   console.log('INSERT ID:',result);
   console.log('#######################');
  });

}

/***********添加表***************/
function creatTable(conn,tablename,callback){
  conn.query('CREATE TABLE '+tablename+' (name varchar(255),password varchar(255),sn varchar(255))',function(err,result){
    if(err){
      console.log('[INSERT ERROR] - ',err.message);
      return err;
    }
    callback(err,result);
    console.log('-------CREATTABLE----------');
    console.log('CREATTABLE ID:',result);
  });

}

var  userAddSql = 'INSERT INTO node_user(name,password,sn) VALUES(?,?,?)';
//var  userAddSql_Params = ['Wilson', '56781234'];
//增 add
function insertFun(conn,name,password,sn,callback){
  conn.query(userAddSql,[name,password,sn],function (err, result) {
          if(err){
           console.log('[INSERT ERROR] - ',err.message);

           return err;
          }
          callback(err,result);
         console.log('-------INSERT----------');
         //console.log('INSERT ID:',result.insertId);
         console.log('INSERT ID:',result);
         console.log('#######################');
  });
}


var userModSql = 'UPDATE node_user SET password = ? WHERE name = ?';
//var userModSql_Params = ['Hello World',99,7];
//改 up
function updateFun(conn,name,password,callback){
  conn.query(userModSql,[name,password],function (err, result) {
     if(err){
           console.log('[UPDATE ERROR] - ',err.message);

           return err;
     }
     callback(err,result);
    console.log('----------UPDATE-------------');
    console.log('UPDATE affectedRows',result.affectedRows);
    console.log('******************************');
  });
}


var  userGetSqlName = 'SELECT * FROM node_user where name = ?';
//查 query
function queryFun(conn,name,callback){
  conn.query(userGetSqlName,[name],function (err, result) {
          if(err){
            console.log('[SELECT ERROR] - ',err.message);

            return err;
          }
          callback(err,result);
         console.log('---------------SELECT----------------');
         console.log(result);
         console.log('$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$');
  });

}

var  userGetSqlSn = 'SELECT * FROM node_user where sn = ?';
//查 query
function queryFunSn(conn,sn,callback){
  conn.query(userGetSqlSn,[sn],function (err, result) {
          if(err){
            console.log('[SELECT ERROR] - ',err.message);

            return err;
          }
          callback(err,result);
         console.log('---------------SELECT----------------');
         console.log(result);
         console.log('$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$');
  });

}

var  userDelSql = 'DELETE FROM node_user WHERE name = ?';
//删除
function delFun(conn,name,callback){
  conn.query(userDelSql,[name],function (err, result) {
          if(err){
            console.log('[DELETE ERROR] - ',err.message);

            return err;
          }
          callback(err,result);
         console.log('-------------DELETE--------------');
         console.log('DELETE affectedRows',result.affectedRows);
         console.log('&&&&&&&&&&&&&&&&&');
  });
}

exports.connect = connectDBMysql;
exports.queryFun = queryFun;
exports.insertFun = insertFun;
exports.delFun = delFun;
exports.updateFun = updateFun;
exports.creatTable = creatTable;
exports.queryFunSn = queryFunSn;
exports.creatdataTable = creatdataTable;
exports.insertdataFun = insertdataFun;
